validator/keymanager/remote-web3signer/keymanager.go

```diff
@@ -77,47 +81,48 @@ func (km *Keymanager) FetchValidatingPublicKeys(ctx context.Context) ([][48]byte
 
 // Sign signs the message by using a remote web3signer server.
 func (km *Keymanager) Sign(ctx context.Context, request *validatorpb.SignRequest) (bls.Signature, error) {
-	if request.Fork == nil {
-		return nil, errors.New("invalid sign request: Fork is nil")
-	}
-
-	signRequestType, err := getSignRequestType(request)
+	signRequest, err := km.getSignRequestJson(request)
 	if err != nil {
 		return nil, err
 	}
 
-	forkData := &v1.Fork{
-		PreviousVersion: hexutil.Encode(request.Fork.PreviousVersion),
-		CurrentVersion:  hexutil.Encode(request.Fork.CurrentVersion),
-		Epoch:           fmt.Sprint(request.Fork.Epoch),
-	}
-	forkInfoData := &v1.ForkInfo{
-		Fork:                  forkData,
-		GenesisValidatorsRoot: hexutil.Encode(km.genesisValidatorsRoot),
-	}
-	aggregationSlotData := &v1.AggregationSlot{Slot: fmt.Sprint(request.AggregationSlot)}
-	web3SignerRequest := SignRequest{
-		Type:            signRequestType,
-		ForkInfo:        forkInfoData,
-		SigningRoot:     hexutil.Encode(request.SigningRoot),
-		AggregationSlot: aggregationSlotData,
-	}
-	return km.client.Sign(ctx, hexutil.Encode(request.PublicKey), &web3SignerRequest)
+	return km.client.Sign(ctx, hexutil.Encode(request.PublicKey), signRequest)
+
 }
 
-// getSignRequestType returns the type of the sign request.
-func getSignRequestType(request *validatorpb.SignRequest) (string, error) {
+// getSignRequestJson returns a json request based on the SignRequest type.
+func (km *Keymanager) getSignRequestJson(request *validatorpb.SignRequest) (SignRequestJson, error) {
```

Comment:
Please do a nil check for safety, otherwise this could panic on line 95 with a nil request

```suggestion
func (km *Keymanager) getSignRequestJson(request *validatorpb.SignRequest) (SignRequestJson, error) {
  if request == nil {
    return nil, errors.New("nil request provided")
  }
```

Reply:
This was resolved but the check was not added. Please add